### PR TITLE
Trading checklist

### PR DESCRIPTION
**Pull request by kaladivo**

### CODE REVIEW

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx]: Move to utils

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx]: MOve to utils

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx:19]: memo?

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx:31]: memo?

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/index.tsx
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/index.tsx:21]: Check preferences? Is this filtered in the tree before?

File: apps/mobile/src/components/ChatDetailScreen/utils.ts
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/utils.ts]: Maybe move this elsewhere?

File: apps/mobile/src/components/ChatDetailScreen/utils.ts
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/utils.ts]: Move this elsewhere

File: apps/mobile/src/components/ChatDetailScreen/utils.ts
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/utils.ts]: Move elsewhere

File: apps/mobile/src/components/ChatDetailScreen/utils.ts
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/utils.ts]: Move elsewhere

File: apps/mobile/src/components/ChatDetailScreen/utils.ts
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/utils.ts]: move elsewhere

File: apps/mobile/src/components/ChatDetailScreen/utils.ts
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/utils.ts:1]: Split this file to subfiles to make it more organized

File: apps/mobile/src/components/ChatDetailScreen/utils.ts
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/utils.ts]: This! Let's put the vexlbot after chat approve message. Should not be that hard.

File: apps/mobile/src/components/Checkbox/index.tsx
Comment (kaladivo) [inline apps/mobile/src/components/Checkbox/index.tsx]: Reconsider naming here.

File: apps/mobile/src/components/TradeChecklistFlow/atoms.ts
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/atoms.ts]: Is this really meant to be deprecated?

File: apps/mobile/src/components/TradeChecklistFlow/atoms/fromChatAtoms.ts
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/atoms/fromChatAtoms.ts:1]: Reconsider file naming. Something seems off....

Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/atoms/fromChatAtoms.ts:1]: Check if atomAtom pattern does not cause any issues.

File: apps/mobile/src/components/TradeChecklistFlow/atoms/mainTradeChecklistStateAtom.ts
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/atoms/mainTradeChecklistStateAtom.ts:7]: This is depreciated right?

File: apps/mobile/src/components/TradeChecklistFlow/atoms/updatesToBeSentAtom.ts
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/atoms/updatesToBeSentAtom.ts:52]: How do we organize this into groups by trading checklist items? For each item in trading checklist there will be atoms to work with updates. Probably better to split into subfiles....

File: apps/mobile/src/components/TradeChecklistFlow/components/AgreeOnTradeDetailsScreen/components/OnlineOrInPersonTrade.tsx
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/AgreeOnTradeDetailsScreen/components/OnlineOrInPersonTrade.tsx]: Consider splitting into multiple files

File: apps/mobile/src/components/TradeChecklistFlow/components/AgreeOnTradeDetailsScreen/components/OnlineOrInPersonTrade.tsx
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/AgreeOnTradeDetailsScreen/components/OnlineOrInPersonTrade.tsx]: Extract callback function? It will be huge

File: apps/mobile/src/components/TradeChecklistFlow/components/AgreeOnTradeDetailsScreen/components/StatusIndicator.tsx
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/AgreeOnTradeDetailsScreen/components/StatusIndicator.tsx:29]: Does pending direction matter?

File: apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/atoms.ts
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/atoms.ts]: This should be in utils function. 
function toStartOfTheDay(milis: UnixMilliseconds): UnixMilliseconds

File: apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/atoms.ts
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/atoms.ts]: Util function for converting DateTime to milliseconds

File: apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/AddTimeOptionsScreen/components/TimeToDropdown.tsx
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/AddTimeOptionsScreen/components/TimeToDropdown.tsx:38]: Maybe explain i=1 it is not clear why it starts at 1

File: apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/OptionsList.tsx
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/OptionsList.tsx:1]: Does it need to be so complicated? It's just a list....

File: apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/PickDateFromSuggestionsScreen.tsx
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/PickDateFromSuggestionsScreen.tsx]: Move to hook, this is really confusing

File: apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/PickTimeFromSuggestions/index.tsx
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/PickTimeFromSuggestions/index.tsx]: Does not work everytime, check if this works here!

File: apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/PickTimeFromSuggestions/index.tsx
Comment (kaladivo) [inline apps/mobile/src/components/TradeChecklistFlow/components/DateAndTimeFlow/components/PickTimeFromSuggestions/index.tsx]: Snímek obrazovky 2023-12-11 v 10 57 52

File: apps/mobile/src/navigationTypes.ts
Comment (kaladivo) [inline apps/mobile/src/navigationTypes.ts:146]: Make the naming more clear. What does it do again?

File: apps/mobile/src/state/chat/atoms/sendTradeChecklistUpdateActionAtom.tsx
Comment (kaladivo) [inline apps/mobile/src/state/chat/atoms/sendTradeChecklistUpdateActionAtom.tsx:44]: Handle lefts?

File: apps/mobile/src/state/chat/atoms/upsertChatForTheirOfferActionAtom.ts
Comment (kaladivo) [inline apps/mobile/src/state/chat/atoms/upsertChatForTheirOfferActionAtom.ts]: Unnecessary spread

File: apps/mobile/src/state/chat/utils/addMessageToChat.ts
Comment (kaladivo) [inline apps/mobile/src/state/chat/utils/addMessageToChat.ts:15]: Is this the right place for the function?

File: apps/mobile/src/state/chat/utils/addMessagesToChats.ts
Comment (kaladivo) [inline apps/mobile/src/state/chat/utils/addMessagesToChats.ts]: Use pipe & optional? Might be more readable

File: apps/mobile/src/state/chat/utils/createNewChatsFromFirstMessages.ts
Comment (kaladivo) [inline apps/mobile/src/state/chat/utils/createNewChatsFromFirstMessages.ts]: unnecessary spread? Not sure. maybe not. If the property is marked as readonly, it is ok not to spread, otherwise spread (just in case...)

File: apps/mobile/src/state/tradeChecklist/atoms/createUpdateTradeChecklistActionAtom.ts
Comment (kaladivo) [inline apps/mobile/src/state/tradeChecklist/atoms/createUpdateTradeChecklistActionAtom.ts]: Default import name should be the same as filename

File: apps/mobile/src/state/tradeChecklist/utils.ts
Comment (kaladivo) [inline apps/mobile/src/state/tradeChecklist/utils.ts:4]: Unit tests?

File: apps/mobile/src/state/tradeChecklist/utils/dateAndTime.ts
Comment (kaladivo) [inline apps/mobile/src/state/tradeChecklist/utils/dateAndTime.ts:1]: unit tests

File: apps/mobile/src/utils/useFocusMemo.ts
Comment (kaladivo) [inline apps/mobile/src/utils/useFocusMemo.ts:1]: describe usage

File: packages/domain/src/general/tradeChecklist.ts
Comment (kaladivo) [inline packages/domain/src/general/tradeChecklist.ts:13]: Also 'waiting for my response based on other side input'. Consider adding another status type

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (kaladivo) [reply apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx]: It seems like it is really specific to vexlBot TradeChecklistDateAndTime, so let's keep it here, just renamed it

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (kaladivo) [reply apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx]: It's specific to functionality in this file. let's just rename it

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (kaladivo) [reply apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx:19]: It's not needed. There is no interaction with this component and we don't expect it to be rerendered that often.

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (kaladivo) [reply apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx:31]: It's not needed. There is no interaction with this component and we don't expect it to be rerendered that often.

File: apps/mobile/src/components/TradeChecklistFlow/atoms/updatesToBeSentAtom.ts
Comment (kaladivo) [reply apps/mobile/src/components/TradeChecklistFlow/atoms/updatesToBeSentAtom.ts:52]: Let's do that when adding a new element.

File: apps/mobile/src/components/TradeChecklistFlow/components/AgreeOnTradeDetailsScreen/components/OnlineOrInPersonTrade.tsx
Comment (kaladivo) [reply apps/mobile/src/components/TradeChecklistFlow/components/AgreeOnTradeDetailsScreen/components/OnlineOrInPersonTrade.tsx]: Let's do that once it really becomes huge

File: apps/mobile/src/navigationTypes.ts
Comment (kaladivo) [reply apps/mobile/src/navigationTypes.ts:146]: But it does makes sense... Lets keep it...

File: apps/mobile/src/state/chat/atoms/sendTradeChecklistUpdateActionAtom.tsx
Comment (kaladivo) [reply apps/mobile/src/state/chat/atoms/sendTradeChecklistUpdateActionAtom.tsx:44]: Handled in call above

File: packages/domain/src/general/tradeChecklist.ts
Comment (kaladivo) [reply packages/domain/src/general/tradeChecklist.ts:13]: Not needed for now. Let's see if this is another UI requirement. We can add it then

File: apps/mobile/src/state/chat/utils/addMessagesToChats.ts
Comment (SamTremko) [inline apps/mobile/src/state/chat/utils/addMessagesToChats.ts]: You need to process not only the last but all

File: apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx
Comment (SamTremko) [inline apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistDateAndTimeView.tsx]: pls intl

File: packages/domain/src/utility/BtcAddress.brand.ts
Comment (SamTremko) [inline packages/domain/src/utility/BtcAddress.brand.ts:5]: add brand

**Review by SamTremko — APPROVED** [review]

**Pull request merged**
